src/app/services/actions/resource-explorer-action-creators.ts

```diff
@@ -44,25 +45,29 @@ export function removeResourcePaths(response: object): AppAction {
 
 export function fetchResources() {
   return async (dispatch: Function, getState: Function) => {
-    try {
-      const { devxApi }: ApplicationState = getState();
-      const resourcesUrl = `${devxApi.baseUrl}/openapi/tree`;
+    const { devxApi }: ApplicationState = getState();
+    const resourcesUrl = `${devxApi.baseUrl}/openapi/tree`;
 
-      const headers = {
-        'Content-Type': 'application/json'
-      };
+    const headers = {
+      'Content-Type': 'application/json'
+    };
 
-      const options: IRequestOptions = { headers };
+    const options: IRequestOptions = { headers };
 
-      dispatch(fetchResourcesPending());
+    dispatch(fetchResourcesPending());
 
+    try {
       const response = await fetch(resourcesUrl, options);
       if (response.ok) {
         const resources = await response.json() as IResource;
         return dispatch(fetchResourcesSuccess(resources));
       }
       throw response;
     } catch (error) {
+      const cachedResources = await resourcesCache.readResources();
```

Comment:
IMO we should be using the cached data in preference to hitting the DevX API every time, it will be much quicker.

Reply:
We only call the api once when the page loads. And there are no use cases where for the app to work a reload is required. 
I'm not sure that there'd be a use case for the preference

Reply:
Our users never hit F5? or visit multiple times a day?

Reply:
that could happen. 
To prefer cache, we'd need to make it able to expire so that we can sometimes get the latest.

Reply:
I've been thinking on this a bit and chatting with @darrelmiller 
We should be looking to ensure that the DevX API provides appropriate cache control headers and not worrying about adding a complex caching mechanism here.

Honestly the less code we write the better in terms of cost of ownership, and it allows us to focus on other areas. 

Do we know how frequently the data in the DevX API is updated?

Reply:
We had a chat with @darrelmiller and we uncovered something I consider a large spanner in the works. The cache control headers  do not work in instances where the client and the api are not in the same domain. 
An alternative exists that we use service workers to try this. However I am not sure whether the code changes will be any less than the minimal work we have done now.

Reply:
@gavinbarron, @darrelmiller  I am more inclined to :
- following your suggesion to prefer cache.
- coming up with a mechanism to expire cache. For resources, we could take advantage of knowledge that the openapi document is refreshed every week. Or simply set cache to expire after 7 days 

Tell me what you think.

Reply:
>The cache control headers do not work in instances where the client and the api are not in the same domain.

O_o

I'm really surprised to hear this because this is literally the situation for resources served from a CDN, those are always cross domain and cache in the browser based on cache control headers.

Is there a way I can take a look at what you're seeing to validate this?

Reply:
@darrelmiller found this when searching. It was a shared screen where we were able to see the results of the query together.

I've been looking this up and it looks like it could be possible even when the client and the API are not in the same domain as long as the client origin is allowed in the API.

Reply:
Yeah, we should be able to set the allowed headers in the response to the options calls from the server:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Headers